packages/flutter/lib/src/services/raw_keyboard_web.dart

```diff
@@ -0,0 +1,198 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+import 'keyboard_key.dart';
+import 'keyboard_maps.dart';
+import 'raw_keyboard.dart';
+
+/// Platform-specific key event data for Web.
+///
+/// See also:
+///
+///  * [RawKeyboard], which uses this interface to expose key data.
+@immutable
+class RawKeyEventDataWeb extends RawKeyEventData {
+  /// Creates a key event data structure specific for Web.
+  ///
+  /// The [keyCode] and [metaState] arguments must not be null.
+  const RawKeyEventDataWeb({
+    @required this.code,
+    @required this.key,
+    this.metaState = modifierNone,
+  })  : assert(code != null),
+        assert(metaState != null);
+
+  /// The `KeyboardEvent.code` corresponding to this event.
+  ///
+  /// See <https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/code>
+  /// for more information.
+  final String code;
+
+  /// The `KeyboardEvent.key` corresponding to this event.
+  ///
+  /// See <https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key>
+  /// for more information.
+  final String key;
+
+  /// The modifiers that were present when the key event occurred.
+  ///
+  /// See `lib/src/engine/keyboard.dart` in the web engine for the numerical
+  /// values of the `metaState`. These constants are also replicated as static
+  /// constants in this class.
+  ///
+  /// See also:
+  ///
+  ///  * [modifiersPressed], which returns a Map of currently pressed modifiers
+  ///    and their keyboard side.
+  ///  * [isModifierPressed], to see if a specific modifier is pressed.
+  ///  * [isControlPressed], to see if a CTRL key is pressed.
+  ///  * [isShiftPressed], to see if a SHIFT key is pressed.
+  ///  * [isAltPressed], to see if an ALT key is pressed.
+  ///  * [isMetaPressed], to see if a META key is pressed.
+  final int metaState;
+
+  @override
+  String get keyLabel => key;
+
+  @override
+  PhysicalKeyboardKey get physicalKey {
+    return kWebToPhysicalKey[code] ?? PhysicalKeyboardKey.none;
+  }
+
+  @override
+  LogicalKeyboardKey get logicalKey {
+    // Look to see if the keyCode is a printable number pad key, so that a
+    // difference between regular keys (e.g. ".") and the number pad version
+    // (e.g. the "." on the number pad) can be determined.
+    final LogicalKeyboardKey numPadKey = kWebNumPadMap[code];
+    if (numPadKey != null) {
+      return numPadKey;
+    }
+
+    // Look to see if the [code] is one we know about and have a mapping for.
+    final LogicalKeyboardKey newKey = kWebToLogicalKey[code];
+    if (newKey != null) {
+      return newKey;
+    }
+
+    // This is a non-printable key that we don't know about, so we mint a new
+    // code with the autogenerated bit set.
+    const int webKeyIdPlane = 0x00900000000;
```

Comment:
@gspencergoog is there a method for assigning these numbers?

Reply:
Yes, see https://github.com/flutter/flutter/blob/master/dev/tools/gen_keycodes/README.md

There's a section in the key id schema that has a table.  The code for web is `0x00800000000`.